### PR TITLE
Setting up front

### PR DESCRIPTION
**Pull request by SaveryIV**
I implement the following features in this branch :

- Created a new Rails API app called 'hello-rails-back-end'.
- The project has a Postgres database set up.
- Created a table for storing your messages and create 5 different greetings.
- Created an API endpoint that selects a random greeting from your table (you will need a controller with an action and rails route).

<hr>

- Created a new React app called 'hello-react-front-end'
- Created the `App `component with react-router.
- Created the `Greeting` component that will display a greeting. Set it up as a route in your App component.
- Created a store, an action and a reducer that will connect to you API endpont to get the random greeting.
- Display the random greeting in your `Greeting` component.

<hr>

- In the README file for the Rails API, added a link to the React app.
- Also, in the README file for the React app, added a link to the Rails API.
- When submitting for a review - use a link to the pull request in your Rails API.
- In the repo for the frontend part - include a PR with changes.

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,25 +1,33 @@
...
+        const data = await response.json();
+        setGreeting(data.greeting); // Assuming the API response has a 'greeting' property
+      } catch (error) {
+        console.log(error);
+      }
+    };
+
+    // Call the fetchGreeting function
+    fetchGreeting();
+  }, []); // The empty dependency array ensures that the effect runs only once
```
Comment (Ibizugbe) [inline src/App.js]: Nice job so far👍🏼, I noticed you already have a greeting component, and you are creating another greeting class in the supposed app component. Just as required, you should have a greeting component and also an `app` component with routes from `react-router` to navigate through your greeting components to render the greetings😉.

image


Kindly ensure your app works properly before proceeding to submit. greetings should be appropriately rendered in the react app.

**Review by Ibizugbe — CHANGES_REQUESTED** [review]
Hi @SaveryIV 👋🏼👋🏼👋🏼, 

request change


Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights✅✅✅

- [x] A greeting component has been created👍🏼.
- [x] Linters are passing👍🏼.
- [x] gitflow is used👍🏼.


## Required Changes ♻️

- [x] Kindly ensure your react app works properly with the rails `API` before requesting a review, to avoid all forms of blockers😉.

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/App.js (hunk 1 above)
Comment (Strangeal) [inline src/App.js]: - [x] Excellent progress on your implementation, and I appreciate your responsiveness to the previous reviewer's feedback! 🚀 However, there are still some important aspects that require attention, specifically related to the project requirements.

   - As outlined in the project specifications, the use of `Redux` for state management is strongly recommended. Although I noticed that you've set up the `reducer`, `action`, and `store`, it appears that you're not leveraging these structures as intended but rather the `API` data is being handled directly in your `useEffect` currently. Below is a screenshot of the project requirement:
   
Screenshot 2023-12-19 at 08 13 07


   - To align with the project's guidelines, I recommend revisiting your implementation and integrating `Redux` more prominently. Utilize the store to manage and distribute the data throughout your frontend application. This adjustment will not only ensure compliance with the specified requirements but also enhance the scalability and maintainability of your codebase. 

Happy coding! 😊👨‍💻🚀

**Review by Strangeal — CHANGES_REQUESTED** [review]
Hi @SaveryIV, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

Alt Text

 ## Highlights
-  Added `model` ✅


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/App.js (hunk 1 above)
Comment (SaveryIV) [reply src/App.js]: I think im using the correct guidelines now, please check and let me know

**Review by omar-muhamad — COMMENTED** [review]
# Invalid ❌

Hi @SaveryIV,

#### While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

#### Your Code Review Request is marked as invalid for this reason:

- [ ] You submitted the wrong PR link. For this project, you should submit the the [backend PR](https://github.com/SaveryIV/hello-rails-back-end/pull/1), not the frontend PR as you did. Please fix that.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **Does not count** into the code reviews limit._

**Review by topeogunleye — APPROVED** [review]
### Hi @SaveryIV, 

# Status: Approved ✔️
Excellent job on making the changes requested by the previous code reviewer. Keep it up 👏

Your project is complete! There is nothing else to say other than... it's time to merge it 💪
Congratulations! 🎉
congratulations
>
## To highlight :champagne: 
- Set up API using the redux store :heavy_check_mark:
 
## [Optional] suggestions:
* Nothing to mention.
 
 _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
 
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @topeogunleye in your question so I can receive the notification.



------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**